### PR TITLE
Fetch Shipping zones from the merchant host

### PR DESCRIPTION
**Pull request by DanReyLop**
Fixes https://github.com/Automattic/wp-calypso/issues/14263

This adds actions & reducers to get the shipping zones list from the WooCommerce host.

I've added a small layer on top of the raw `wpcom.js` library, if you folks have nothing negative to say about it, I think we should use it all over WooCommerce.

**Note**: Since error handling is still something that's being debated & implemented, I've omitted it from this PR.

**To test**: Run `make test`, see that the tests pass.

### CODE REVIEW

File: client/extensions/woocommerce/state/wc-api/request.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const addURLParam = ( url, param ) => url + ( -1 !== url.indexOf( '?' ) ? '?' : '&' ) + param;
+
+/**
+ * Higher-level layer on top of the WPCOM.JS library, made specifically for making requests to a
+ * Jetpack-connected WooComemrce site.
+ * @param {Number} siteId Site ID to make the request to
+ * @return {Object} An object with the properties "get", "post", "put" and "del", which are functions to
+ * make an HTTP GET, POST, PUT and DELETE request, respectively.
+ */
+export default ( siteId ) => ( {
```
Comment (coderkevin) [inline client/extensions/woocommerce/state/wc-api/request.js]: I like this implementation. Good work!

File: client/extensions/woocommerce/state/wc-api/shipping-zones/reducer.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+} from '../../action-types';
+
+export const LOADING = 'LOADING';
+
+// TODO: Handle error
+
+export default {
+	[ WOOCOMMERCE_API_FETCH_SHIPPING_ZONES ]: ( state ) => {
+		return { ...state,
+			shippingZones: LOADING,
```
Comment (coderkevin) [inline client/extensions/woocommerce/state/wc-api/shipping-zones/reducer.js]: This is an interesting approach. I've usually just held an empty set until data existed.

Comment (DanReyLop) [inline client/extensions/woocommerce/state/wc-api/shipping-zones/reducer.js]: The problem with that is that an empty set is a valid API response, so if you want to track if the collection is being loaded, you would need a separate field.

**Comment by DanReyLop** [comment]
Thenks for the review! I'll wait to merge this until https://github.com/Automattic/wp-calypso/pull/13542 is merged (as we were warned).

File: client/extensions/woocommerce/state/wc-api/request.js (hunk 1 above)
Comment (belcherj) [inline client/extensions/woocommerce/state/wc-api/request.js]: I agree

File: client/extensions/woocommerce/state/wc-api/shipping-zones/selectors.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ */
+import { get, isArray } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { LOADING } from './reducer';
+
+const getRawShippingZones = ( state, siteId ) => get( state, [ 'extensions', 'woocommerce', 'wcApi', siteId, 'shippingZones' ] );
```
Comment (belcherj) [inline client/extensions/woocommerce/state/wc-api/shipping-zones/selectors.js]: this line is really long

**Pull request merged**